src/tools/rbd_mirror/image_replayer/PrepareLocalImageRequest.cc

```diff
@@ -124,9 +124,16 @@ void PrepareLocalImageRequest<I>::handle_get_mirror_info(int r) {
     return;
   }
 
-  // TODO save current mirror state to determine if we should
-  // delete a partially formed image
-  // (e.g. MIRROR_IMAGE_STATE_CREATING/DELETING)
+  if (m_mirror_image.state == cls::rbd::MIRROR_IMAGE_STATE_CREATING) {
+    dout(5) << "remote image mirroring is still creating, issuing a removal"
+            << dendl;
+    disable_mirror_image();
```

Comment:
I am not quite understand this log message? Why it is saying about the remote image and why it is still creating?

As I understand if we end up in this situation it means that the previous rbd-mirror run was interrupted at the moment when it was creating the image (local, non-primary). And we need to do the cleanup. Just removing the mirror image metadata seems not enough to me, because the real image header (and other metadata) could already been created and we need to remove it. What about calling `image_deleter::TrashMoveRequest` here instead? It would try to remove the image if it is already created.

Reply:
Oups yes for the message it should be better now. I am also using the TrashMoveRequest now.